src/test/java/htsjdk/samtools/util/BlockCompressedInputStreamTest.java

```diff
@@ -93,80 +93,85 @@ public void available_should_return_number_of_bytes_left_in_current_block() thro
         final List<String> linesWritten = new ArrayList<>();
-        final BlockCompressedOutputStream bcos = new BlockCompressedOutputStream(tempFile, 5);
-        String s = "Hi, Mom!\n";
-        bcos.write(s.getBytes()); //Call 1
-        linesWritten.add(s);
-        s = "Hi, Dad!\n";
-        bcos.write(s.getBytes()); //Call 2
-        linesWritten.add(s);
-        bcos.flush();
-        final StringBuilder sb = new StringBuilder(BlockCompressedStreamConstants.DEFAULT_UNCOMPRESSED_BLOCK_SIZE * 2);
-        s = "1234567890123456789012345678901234567890123456789012345678901234567890\n";
-        while (sb.length() <= BlockCompressedStreamConstants.DEFAULT_UNCOMPRESSED_BLOCK_SIZE) {
-            sb.append(s);
+        try ( final BlockCompressedOutputStream bcos = new BlockCompressedOutputStream(tempFile, 5) ) {
+            String s = "Hi, Mom!\n";
+            bcos.write(s.getBytes()); //Call 1
             linesWritten.add(s);
-        }
-        bcos.write(sb.toString().getBytes()); //Call 3
-        bcos.close();
-
-        final InflaterFactory myInflaterFactory = new InflaterFactory() {
-            public Inflater makeInflater(final boolean gzipCompatible) {
-                return new MyInflater(gzipCompatible);
+            s = "Hi, Dad!\n";
+            bcos.write(s.getBytes()); //Call 2
+            linesWritten.add(s);
+            bcos.flush();
+            final StringBuilder sb = new StringBuilder(BlockCompressedStreamConstants.DEFAULT_UNCOMPRESSED_BLOCK_SIZE * 2);
+            s = "1234567890123456789012345678901234567890123456789012345678901234567890\n";
+            while ( sb.length() <= BlockCompressedStreamConstants.DEFAULT_UNCOMPRESSED_BLOCK_SIZE ) {
+                sb.append(s);
+                linesWritten.add(s);
             }
-        };
-
-        // test catching null InflaterFactory
-        try {
-            BlockGunzipper.setDefaultInflaterFactory(null);
-            Assert.fail("Did not catch 'null' InflaterFactory");
+            bcos.write(sb.toString().getBytes()); //Call 3
         }
-        catch (java.lang.IllegalArgumentException e) { /* expected */ }
+        return linesWritten;
+    }
+
+    @DataProvider(name = "customInflaterInput")
+    public Object[][] customInflateInput() throws IOException {
+        final File tempFile = File.createTempFile("testCustomInflater.", ".bam");
+        tempFile.deleteOnExit();
+        final List<String> linesWritten = writeTempBlockCompressedFileForInflaterTest(tempFile);
 
-        BlockGunzipper.setDefaultInflaterFactory(myInflaterFactory);
+        final InflaterFactory countingInflaterFactory = new CountingInflaterFactory();
+        BlockGunzipper.setDefaultInflaterFactory(countingInflaterFactory);
 
         return new Object[][]{
                 // use default InflaterFactory
-                {new BlockCompressedInputStream(new FileInputStream(tempFile), false), linesWritten, 4},
```

Comment:
although, since it's creating a temp file anyway, which is probably the most dangerous part, I'm not sure how much this is actually buying you.